examples/cluster/plot_segmentation_toy.py

```diff
@@ -51,8 +51,9 @@
 circle3 = (x - center3[0]) ** 2 + (y - center3[1]) ** 2 < radius3**2
 circle4 = (x - center4[0]) ** 2 + (y - center4[1]) ** 2 < radius4**2
 
-# #############################################################################
-# 4 circles
+# %%
+# 4 Circles
+# ---------
 img = circle1 + circle2 + circle3 + circle4
 
 # We use a mask that limits to the foreground: the problem that we are
```

Comment:
Switching the position of the code and the comment here will cause the comment to be rendered as text. Given that the comments in this heading are rather descriptive, I think this would be to the reader's advantage.

I would further suggest that a new cell delimiter (i.e. `# %%`) be placed above each comment in this section so that all explanations are rendered as text here. For example, the next comment would become:
```diff
-# Convert the image into a graph with the value of the gradient on the
-# edges.

+# %%
+# Convert the image into a graph with the value of the gradient on the
+# edges.
```

Personally, I would also make a few minor revisions to some of the comments. Specifically:
```diff
-# Take a decreasing function of the gradient: we take it weakly
-# dependent from the gradient the segmentation is close to a voronoi

+# %%
+# Take a decreasing function of the gradient resulting in a segmentation
+# that is close to a Voronoi partition.
```

```diff
-# Force the solver to be arpack, since amg is numerically
-# unstable on this example

+# %%
+# Here we perform spectral clustering using the arpack solver since amg is
+# numerically unstable on this example. We then plot the results.
```